pkg/misc/depversion/depversion.go

```diff
@@ -332,7 +335,7 @@ func getConstraint(s string) (string, error) {
 	}
 	// check for 1.x minor and patch versions
 	if isSemVerWildcard(s) {
-		_, major, minor, _, _, _, err := parseWildcardSemver(s)
+		_, major, minor, _, _, _, err := parseWildcardSemver(strings.TrimPrefix(s, "~"))
```

Comment:
Follow the same pattern already found in this file for consistency:
```
version := strings.TrimPrefix(s, "~")
_, major, minor, _, _, _, err := parseWildcardSemver(version)
```
Or change all the other cases to do the trim inline when passed to the parse function().

Also, add a comment here that we are not expecting ~ in all cases, but when found we just ignore it and assume the wildcard handling is correct.

Reply:
I have got the code to ignore any tildes that are seen by the wildcard check.